dataclients/kubernetes/routegroup.go

```diff
@@ -478,7 +480,34 @@ func splitHosts(hosts []string, domains []string) ([]string, []string) {
 	return internalHosts, externalHosts
 }
 
-func (r *routeGroups) convert(s *clusterState, df defaultFilters, loggingEnabled bool) ([]*eskip.Route, error) {
+// addRouteGroupTLS compares the RouteGroup host list and the RouteGroup TLS host list
+// and adds the TLS secret to the registry if a match is found.
+func (r *routeGroups) addRouteGroupTLS(ctx *routeGroupContext, tls *definitions.RouteTLSSpec) {
+	// Host in the tls section need to explicitly match the host in the RouteGroup
+	hostlist := compareStringList(tls.Hosts, ctx.routeGroup.Spec.UniqueHosts())
```

Comment:
I think what we want here is to check if `tls.Hosts` is a subset of `ctx.routeGroup.Spec.UniqueHosts()`. 
`compareStringList` returns all items from `tls.Hosts` that are also part of `ctx.routeGroup.Spec.UniqueHosts()`.
So it reduces the hostlist.

Did you do this by intention and I miss something?

Reply:
We only want to add certs which have a match from from tls.Hosts and the routeGroup hosts, so reducing the list is desired. If there are no matches we log and return, but if there are matches, we use the reduced list to add the matching tls certs to the registry. 

We could log the non-matching hosts in tls.hosts to help debugging a miss configuration. What do you think?

Reply:
Yes logging makes sense.
I wonder in case of an update that is applied wrong, what we want to have as behavior. For example a timeline like this:
1. rg foo has tls.hosts foo.example, bar.example and there is a referenced secret that has only these two names foo.example, bar.example as hosts in SAN
2. rg was updated to add baz.example and remove bar.example in tls.hosts

The current behavior would drop bar.example.

Should this better stop updating it, such that bar.example is not replaced by baz.example ?!

I am thinking of migrations, but can not come up with an idea how this could break less with either case, so from my side we can also just log it.

@AlexanderYastrebov wdyt?

Reply:
Maybe this should be a part of validator and invalid routegroup should be rejected. We already use validator in webhook and dataclient.

Reply:
The same is for Ingress - we plan to add ingress validator to dataclient, see https://github.com/zalando/skipper/pull/2757

Reply:
Is this something we could handle in a separate PR to keep the scope limited? Happy to work on getting in validator for both Ingress and RouteGroup.

Reply:
Lets have it like it is now (log mismatch) and add tls validation for ingress and routegroup as a future improvement (makes sense after we land https://github.com/zalando/skipper/pull/2757).
@MustafaSaber FYI